Sources/NIOCore/ChannelHandlers.swift

```diff
@@ -189,11 +189,22 @@ public final class IdleStateHandler: ChannelDuplexHandler, RemovableChannelHandl
     private var scheduledReaderTask: Optional<Scheduled<Void>>
     private var scheduledWriterTask: Optional<Scheduled<Void>>
     private var scheduledAllTask: Optional<Scheduled<Void>>
+    private var now: () -> NIODeadline
+    
+    public convenience init(readTimeout: TimeAmount? = nil, writeTimeout: TimeAmount? = nil, allTimeout: TimeAmount? = nil) {
```

Comment:
I didn't want to open the `now` parameter to the public so I opted in for a public convenience initializer here with the same public API as before.